src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_Wasm/WasmReadyToRunGenericHelperNode.cs

```diff
@@ -0,0 +1,90 @@
+                        {
+                            encoder.Builder.EmitReloc(factory.HelperEntrypoint(HelperEntrypoint.EnsureClassConstructorRunAndReturnNonGCStaticBase), RelocType.IMAGE_REL_BASED_REL32);
+                        }
+                    }
+                    break;
+
+                case ReadyToRunHelperId.GetGCStaticBase:
+                    {
+                        MetadataType target = (MetadataType)_target;
+
+                        if (factory.PreinitializationManager.HasLazyStaticConstructor(target))
+                        {
+                            encoder.Builder.EmitReloc(factory.HelperEntrypoint(HelperEntrypoint.EnsureClassConstructorRunAndReturnGCStaticBase), RelocType.IMAGE_REL_BASED_REL32);
+                        }
+                    }
+                    break;
+
+                case ReadyToRunHelperId.GetThreadStaticBase:
+                    {
+                        MetadataType target = (MetadataType)_target;
+
+                        ISymbolNode helperEntrypoint;
+                        if (factory.PreinitializationManager.HasLazyStaticConstructor(target))
+                        {
+                            helperEntrypoint = factory.HelperEntrypoint(HelperEntrypoint.EnsureClassConstructorRunAndReturnThreadStaticBase);
+                        }
+                        else
+                        {
+                            helperEntrypoint = factory.HelperEntrypoint(HelperEntrypoint.GetThreadStaticBaseForType);
+                        }
+
+                        encoder.Builder.EmitReloc(helperEntrypoint, RelocType.IMAGE_REL_BASED_REL32);
+                    }
+                    break;
+
+                case ReadyToRunHelperId.DelegateCtor:
+                    {
+                        var target = (DelegateCreationInfo)_target;
+
+                        if (target.Thunk != null)
+                        {
+                            encoder.Builder.EmitReloc(target.Thunk, RelocType.IMAGE_REL_BASED_REL32);
+                        }
+                        else
+                        {
+                            Debug.Assert(target.Constructor.Method.Signature.Length == 2);
+                        }
+
+                        encoder.Builder.EmitReloc(target.Constructor, RelocType.IMAGE_REL_BASED_REL32);
+                    }
+                    break;
+
+                // These are all simple: do dependencies
+                case ReadyToRunHelperId.TypeHandle:
+                case ReadyToRunHelperId.MethodHandle:
+                case ReadyToRunHelperId.FieldHandle:
+                case ReadyToRunHelperId.MethodDictionary:
+                case ReadyToRunHelperId.MethodEntry:
+                case ReadyToRunHelperId.VirtualDispatchCell:
+                case ReadyToRunHelperId.DefaultConstructor:
+                case ReadyToRunHelperId.ObjectAllocator:
+                case ReadyToRunHelperId.TypeHandleForCasting:
+                    break;
```

Comment:
Might be useful to just have this report the dependencies now - otherwise there will be extra dependency graph issues to investigate and root cause to this line of code when the `!TargetArchIsWasm()` workaround is removed.

Reply:
Are you refering to these dependencies:

https://github.com/yowl/runtimelab/blob/0480a1fc0733642f9116ad7d5fc2b44e2f40a4d4/src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/ILToLLVMImporter.cs#L2440

?

Reply:
These report dependencies that are needed for what `CallGenericHelper` does.

E.g. if there's a call to `Is<object>(new object());` for this:

```csharp
static bool Is<T>(object o) => o is T;
```

How does the generic dictionary for `Is<object>` look like after scanning? I expect it will be empty.

The method body of `Is<__Canon>` will be scanned as having a call to the generic lookup helper to get the T, but the generic lookup helper won't report any dependency, because we only do ` break` at this point.

Reply:
Sorry, I'm still missing something here.  The x64 version doesn't add any dependencies for these via the encoder, why is LLVM/Wasm different?

Reply:
Ah, you're right, it doesn't. Well except for this one but that's niche:

https://github.com/dotnet/runtimelab/blob/04b092003db5ba207d4fa3f3becb7f01828bf16c/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_X64/X64ReadyToRunGenericHelperNode.cs#L49

Reply:
Ah, because the branch is not up to date regards merging and I don't have `ThrowUnavailableType` yet.  I've added a TODO, which is the best I can do right now I suppose.